server_api.py

```diff
@@ -307,16 +307,15 @@ def get_prefer_news(db, prefer_data_type):
         prefer_str = prefer_str[:-1]
         prefer_str = prefer_str + ")"
         #find two
-        sql = "SELECT * FROM "\
-        +"(SELECT title, serial_number,data_type FROM news_QR_code where is_delete=0 and data_type IN " + prefer_str \
-        +" ORDER BY upload_time DESC LIMIT 10) as data ORDER BY RAND() LIMIT 2"
-        pure_result = db.query(sql)
+        with NewsQRCodeDao() as newsQRCodeDao:
+            pure_result = newsQRCodeDao.getNews(preferStr=prefer_str)
         #reshape output data
         for num2 in range(len(pure_result)):
-            type_dir = db.query('select type_dir from data_type where type_id = %s' % pure_result[num2][2])[0]
+            with DataTypeDao() as dataTypeDao:
+                type_dir = dataTypeDao.getTypeDir(typeId=pure_result[num2][2])
```

Comment:
I think one PR should include one purpose for change.
The above change can open another PR.